dirty_cat/similarity_encoder.py

```diff
@@ -2,9 +2,7 @@
 from scipy import sparse
 
 from sklearn.feature_extraction.text import CountVectorizer, HashingVectorizer
-from sklearn.preprocessing._encoders import _BaseEncoder
-from sklearn.utils import check_array
-
+from sklearn.preprocessing._encoders import OneHotEncoder
```

Comment:
Can you import from sklearn.preprocessing rather than the private module.

Reply:
sure !